analyzers/src/SonarAnalyzer.CSharp/Rules/UseAwaitableMethod.cs

```diff
@@ -0,0 +1,164 @@
+                    codeBlockStart.RegisterNodeAction(nodeContext =>
+                    {
+                        var invocationExpression = (InvocationExpressionSyntax)nodeContext.Node;
+
+                        var awaitableAlternatives = FindAwaitableAlternatives(wellKnownExtensionMethodContainer, codeBlock, invocationExpression,
+                            nodeContext.SemanticModel, nodeContext.ContainingSymbol, nodeContext.Cancel);
+                        if (awaitableAlternatives.Any())
+                        {
+                            nodeContext.ReportIssue(Rule, invocationExpression, awaitableAlternatives.First().Name);
+                        }
+                    }, SyntaxKind.InvocationExpression);
+                }
+            });
+        });
+
+    private static WellKnownExtensionMethodContainer BuildWellKnownExtensionMethodContainers(Compilation compilation)
+    {
+        var wellKnownExtensionMethodContainer = new WellKnownExtensionMethodContainer();
+        var queryable = compilation.GetTypeByMetadataName(KnownType.System_Linq_Queryable);
+        var enumerable = compilation.GetTypeByMetadataName(KnownType.System_Linq_Enumerable);
+
+        if (compilation.GetTypeByMetadataName(KnownType.Microsoft_EntityFrameworkCore_EntityFrameworkQueryableExtensions) is { } entityFrameworkQueryableExtensions)
+        {
+            wellKnownExtensionMethodContainer.Add(queryable, entityFrameworkQueryableExtensions);
+            wellKnownExtensionMethodContainer.Add(enumerable, entityFrameworkQueryableExtensions);
+        }
+        if (compilation.GetTypeByMetadataName(KnownType.Microsoft_EntityFrameworkCore_RelationalQueryableExtensions) is { } relationalQueryableExtensions)
+        {
+            wellKnownExtensionMethodContainer.Add(queryable, relationalQueryableExtensions);
+            wellKnownExtensionMethodContainer.Add(enumerable, relationalQueryableExtensions);
+        }
+
+        return wellKnownExtensionMethodContainer;
+    }
+
+    private static IEnumerable<IMethodSymbol> GetMethodSymbolsInScope(string methodName, WellKnownExtensionMethodContainer wellKnownExtensionMethodContainer,
+        ITypeSymbol invokedType, ITypeSymbol methodContainer) =>
+        ((ITypeSymbol[])[.. invokedType.GetSelfAndBaseTypes(), .. WellKnownExtensionMethodContainer(wellKnownExtensionMethodContainer, methodContainer), methodContainer])
+            .Distinct()
+            .SelectMany(x => x.GetMembers(methodName))
+            .OfType<IMethodSymbol>();
+
+    private static IEnumerable<INamedTypeSymbol> WellKnownExtensionMethodContainer(WellKnownExtensionMethodContainer lookup, ITypeSymbol invokedType) =>
+        lookup.TryGetValue(invokedType, out var extensionMethodContainer)
+        ? extensionMethodContainer
+        : [];
+
+    private static ImmutableArray<ISymbol> FindAwaitableAlternatives(WellKnownExtensionMethodContainer wellKnownExtensionMethodContainer, SyntaxNode codeBlock,
```

Comment:
You are right, this is because of the CodeFix. I was thinking about, also looking for "overloads" that take an additional `CancelationToken` to support cases like this: [Task.Wait()](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-8.0) but there is only [Task.WaitAsync(CancellationToken)](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-8.0#system-threading-tasks-task-waitasync(system-threading-cancellationtoken))